beauty-consulting-app/app/Pages/ProfileView.js

```diff
@@ -1,119 +1,437 @@
+			// When exiting edit mode, submit the changes
+			await submitEdit();
+		}
+	};
+
+	// Function to submit the edited user information
+	const submitEdit = async () => {
+		// Preserve existing values if new values are empty
+		const updatedName = name !== "" ? name : userContext.name;
+		const updatedGender = gender !== "" ? gender : userContext.gender;
+		const updatedAllergies =
+			allergies !== "" ? allergies : userContext.allergies;
+		const updatedConcerns =
+			concerns !== "" ? concerns : userContext.concerns;
+
+		setLoading(true);
+
+		const req = {
+			username: userContext.username,
+			name: updatedName,
+			age: userContext.age, // Assuming age is not editable here
+			gender: updatedGender,
+			phoneNumber: userContext.phoneNumber, // Assuming phoneNumber is not editable here
+			email: userContext.email, // Assuming email is not editable here
+			hairDetails: userContext.hairDetails, // Assuming hairDetails are handled elsewhere
+			allergies: updatedAllergies,
+			concerns: updatedConcerns,
+		};
+
+		try {
+			const apiURL =
+				Constants.manifest?.extra?.apiUrl ||
+				process.env.EXPO_PUBLIC_API_URL;
+			if (!apiURL) {
+				console.error("apiURL not defined");
+				Alert.alert("Error", "API URL is not defined.");
+				setLoading(false);
+				return;
 			}
+			const res = await axios.put(
+				`${apiURL}/client/${userContext.username}`,
+				req
+			);
+			console.log("update successful: ", res.data);
 
-			// update user context for rest of session
-			userContext = userContext.updateUserContext({
+			// Update UserContext with the new information
+			userContext.updateUserContext({
 				username: userContext.username,
-				name: name,
-				age: age,
-				gender: gender,
+				name: updatedName,
+				age: userContext.age,
+				gender: updatedGender,
 				phoneNumber: userContext.phoneNumber,
 				email: userContext.email,
 				hairDetails: userContext.hairDetails,
-				allergies: allergies,
-				concerns: concerns,
-				updateUserContext: userContext.updateUserContext,
+				allergies: updatedAllergies,
+				concerns: updatedConcerns,
 			});
+
+			Alert.alert("Success", "Profile updated successfully.");
+		} catch (error) {
+			console.error("Error with request: ", error);
+			Alert.alert(
+				"Error",
+				"Failed to update profile. Please try again later."
+			);
+		} finally {
+			setLoading(false);
 		}
-		setIsEdit(!isEdit);
 	};
 
+	// Function to handle account deletion
 	const deleteAccount = async () => {
 		// TODO: update with a separate flow for password validation after backend rework
 
+		setLoading(true);
+
 		try {
 			console.log(userContext);
-			const res = api.delete(`/client/${userContext.username}`);
+			const apiURL =
+				Constants.manifest?.extra?.apiUrl ||
+				process.env.EXPO_PUBLIC_API_URL;
+			if (!apiURL) {
+				console.error("apiURL not defined");
+				Alert.alert("Error", "API URL is not defined.");
+				setLoading(false);
+				return;
+			}
```

Comment:
See #89, this should all not be necessary.

Reply:
Resolved.